### PR TITLE
NETOBSERV-2397 QE: Automate test for auto-detect privilege run feature in network observability CLI

### PR DESCRIPTION
**Pull request by oliver-smakal**
## Description

Adds tests to verify correct privilege settings as introduced in: [NETOBSERV-2262](https://issues.redhat.com//browse/NETOBSERV-2262) 

Please also verify that polarion test case is created correctly: https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-84801) is created correctly.
## Dependencies

n/a
#
## Checklist

If you are not familiar with our processes or don't know what to answer in the list below, let us know in a comment: the maintainers will take care of that.

* [x] Is this PR backed with a JIRA ticket? If so, make sure it is written as a title prefix _(in general, PRs affecting the NetObserv/Network Observability product should be backed with a JIRA ticket - especially if they bring user facing changes)._
* [ ] Does this PR require product documentation?
  * [ ] If so, make sure the JIRA epic is labelled with "documentation" and provides a description relevant for doc writers, such as use cases or scenarios. Any required step to activate or configure the feature should be documented there, such as new CRD knobs.
* [ ] Does this PR require a product release notes entry?
  * [ ] If so, fill in "Release Note Text" in the JIRA.
* [x] Is there anything else the QE team should know before testing? E.g: configuration changes, environment setup, etc.
  * [x] If so, make sure it is described in the JIRA ticket.
    * polarion test case needs to be also verified, follow link in the description above 
* QE requirements (check 1 from the list):
  * [x] Standard QE validation, with pre-merge tests unless stated otherwise.
  * [ ] Regression tests only (e.g. refactoring with no user-facing change).
  * [ ] No QE (e.g. trivial change with high reviewer's confidence, or per agreement with the QE team).

### CODE REVIEW

File: e2e/integration-tests/integration_test.go [hunk 1]
```diff
@@ -227,4 +228,72 @@ var _ = g.Describe("NetObserv CLI e2e integration test suite", g.Ordered, func()
...
+
+			{
+				desc: "Verifying `oc netobserv flows --drops --enable_network_events=true` runs as privileged",
+				cliArgs: []string{"flows", "--enable_network_events=true"},
+				matcher: o.BeTrue(),
+			},
+			{
+				desc: "Verifying `oc netobserv flows --drops --enable_network_events=true --privileged=false` is overwritten and runs as privileged",
+				cliArgs: []string{"flows", "--enable_network_events=true", "--privileged=false"},
+				matcher: o.BeTrue(),
```
[the remark this answers is not in the document]
Comment (oliver-smakal) [reply e2e/integration-tests/integration_test.go:272]: However the privilege adjustment is coded on a singe feature level - I think it cannot be said that if it works for `--drops`  that is will work the other ones such as `--enable_network_events`.

But maybe it is not worth testing it.

File: e2e/integration-tests/integration_test.go [hunk 2]
```diff
@@ -227,4 +228,72 @@ var _ = g.Describe("NetObserv CLI e2e integration test suite", g.Ordered, func()
...
+			},
+
+			{
+				desc: "Verifying `oc netobserv flows --drops` runs as privileged",
+				cliArgs: []string{"flows", "--drops"},
+				matcher: o.BeTrue(),
+			},
+			{
+				desc: "Verifying `oc netobserv flows --drops --privileged=false` is overwritten and runs as privileged",
```
[the remark this answers is not in the document]
Comment (oliver-smakal) [reply e2e/integration-tests/integration_test.go:259]: The reason I included this, is that it can break the interface if behavior changes - but considering it is not intended to be used in a scripting manner, dropping this should not be a big deal.

[the remark this answers is not in the document]
Comment (oliver-smakal) [reply e2e/integration-tests/integration_test.go]: resolved in commit [df2c2ba1d8c0df19bd10d6b65ec99f527ed33561](https://github.com/oliver-smakal/network-observability-cli/commit/df2c2ba1d8c0df19bd10d6b65ec99f527ed33561)

File: e2e/integration-tests/integration_test.go [hunk 3]
```diff
@@ -227,4 +228,72 @@ var _ = g.Describe("NetObserv CLI e2e integration test suite", g.Ordered, func()
...
+				matcher: o.BeTrue(),
+			},
+			{
+				desc: "Verifying `oc netobserv flows --drops --enable_network_events=true --privileged=false` is overwritten and runs as privileged",
+				cliArgs: []string{"flows", "--enable_network_events=true", "--privileged=false"},
+				matcher: o.BeTrue(),
+			},
+		}
+
+		for _, t := range tests {
```
[the remark this answers is not in the document]
Comment (oliver-smakal) [reply e2e/integration-tests/integration_test.go:262]: > Eg, from the current g.It() I would move the new test-case to g.Describe() and each scenario into g.It. If that will not cause any problems - I will go ahead and do it like that.

Confirmed via call, implemented.

**Comment by oliver-smakal** [comment]
> @oliver-smakal - could you add local run of tests in PR description? We see some flakiness in CI

It does that locally the integration tests are failing and taking quite long. I am running it again rosa 4.19 cluster, but that should not be an issue, right?

Example logs:
Log1:
```
osmakal@osmakal-thinkpadp1gen7:~/Repos/cli$ go test -p 1 -timeout 45m -v -mod vendor ./e2e/integration-tests/...
=== RUN   TestIntegrationTests
Running Suite: IntegrationTests Suite - /home/osmakal/Repos/cli/e2e/integration-tests
=====================================================================================
Random Seed: 1758792960

Will run 7 of 7 specs
time="2025-09-25T11:36:00+02:00" level=info msg="Running command" arg="[flows --query=SrcK8S_Namespace=~\"openshift-monitoring\" --copy=true --max-bytes=500000 --max-time=1m]" cmd=/usr/local/bin/oc-netobserv component=integration_test
time="2025-09-25T11:37:23+02:00" level=info msg="Cleaning up..." component=integration_test
time="2025-09-25T11:37:23+02:00" level=info msg="Running command" arg="[cleanup]" cmd=/usr/local/bin/oc-netobserv component=integration_test
------------------------------
• [FAILED] [89.557 seconds]
NetObserv CLI e2e integration test suite [It] OCP-73458: Verify regexes filters are applied [Regexes]
/home/osmakal/Repos/cli/e2e/integration-tests/integration_test.go:88

  [FAILED] Flow {openshift-insights} does not meet regexes condition SrcK8S_Namespace=~openshift-monitoring
  Expected
      <bool>: true
  to be false
  In [It] at: /home/osmakal/Repos/cli/e2e/integration-tests/integration_test.go:122 @ 09/25/25 11:37:23.75
------------------------------
SSSSSS

Summarizing 1 Failure:
  [FAIL] NetObserv CLI e2e integration test suite [It] OCP-73458: Verify regexes filters are applied [Regexes]
  /home/osmakal/Repos/cli/e2e/integration-tests/integration_test.go:122

Ran 1 of 7 Specs in 89.559 seconds
FAIL! -- 0 Passed | 1 Failed | 0 Pending | 6 Skipped
--- FAIL: TestIntegrationTests (89.56s)
FAIL
FAIL    github.com/netobserv/network-observability-cli/e2e/integration-tests    89.573s
FAIL
```
Log2:
```
osmakal@osmakal-thinkpadp1gen7:~/Repos/cli$ go test -p 1 -timeout 45m -v -mod vendor ./e2e/integration-tests/...
=== RUN   TestIntegrationTests
Running Suite: IntegrationTests Suite - /home/osmakal/Repos/cli/e2e/integration-tests
=====================================================================================
Random Seed: 1758793075

Will run 6 of 6 specs
time="2025-09-25T11:37:55+02:00" level=info msg="Running command" arg="[flows --sampling=1 --copy=true --max-bytes=500000 --max-time=1m]" cmd=/usr/local/bin/oc-netobserv component=integration_test
time="2025-09-25T11:39:19+02:00" level=info msg="Cleaning up..." component=integration_test
time="2025-09-25T11:39:19+02:00" level=info msg="Running command" arg="[cleanup]" cmd=/usr/local/bin/oc-netobserv component=integration_test
•time="2025-09-25T11:39:25+02:00" level=info msg="Running command" arg="[flows --exclude_interfaces=genev_sys_6081 --copy=true --max-bytes=500000 --max-time=1m]" cmd=/usr/local/bin/oc-netobserv component=integration_test
time="2025-09-25T11:40:49+02:00" level=info msg="Cleaning up..." component=integration_test
time="2025-09-25T11:40:49+02:00" level=info msg="Running command" arg="[cleanup]" cmd=/usr/local/bin/oc-netobserv component=integration_test

------------------------------
• [FAILED] [90.059 seconds]
NetObserv CLI e2e integration test suite [It] OCP-82597: Verify excluded interface genev_sys_6081 does not appear in captured flows [ExcludeInterface]
/home/osmakal/Repos/cli/e2e/integration-tests/integration_test.go:166

  [FAILED] Flow should not contain excluded interface genev_sys_6081, but found it in interfaces: [genev_sys_6081 b9949d7f76b9f2c]
  Expected
      <string>: genev_sys_6081
  not to equal
      <string>: genev_sys_6081
  In [It] at: /home/osmakal/Repos/cli/e2e/integration-tests/integration_test.go:198 @ 09/25/25 11:40:49.189
------------------------------
SSSS

Summarizing 1 Failure:
  [FAIL] NetObserv CLI e2e integration test suite [It] OCP-82597: Verify excluded interface genev_sys_6081 does not appear in captured flows [ExcludeInterface]
  /home/osmakal/Repos/cli/e2e/integration-tests/integration_test.go:198

Ran 2 of 6 Specs in 179.926 seconds
FAIL! -- 1 Passed | 1 Failed | 0 Pending | 4 Skipped
--- FAIL: TestIntegrationTests (179.93s)
FAIL
FAIL    github.com/netobserv/network-observability-cli/e2e/integration-tests    179.941s
FAIL
```

trying investigate further

**Review by memodi — APPROVED** [review]
I ran it just by filtering new test in PR:

```
ginkgo --label-filter=Privileges run e2e/integration-tests

================================================================================================================================
Random Seed: 1758817130

Will run 3 of 8 specs
SSSSStime="2025-09-25T12:18:56-04:00" level=info msg="Starting command" arg="[flows]" cmd=/usr/local/bin/oc-netobserv component=integration_test
time="2025-09-25T12:19:26-04:00" level=info msg="Cleaning up..." component=integration_test
time="2025-09-25T12:19:26-04:00" level=info msg="Running command" arg="[cleanup]" cmd=/usr/local/bin/oc-netobserv component=integration_test
•time="2025-09-25T12:20:13-04:00" level=info msg="Starting command" arg="[flows --privileged=true]" cmd=/usr/local/bin/oc-netobserv component=integration_test
time="2025-09-25T12:20:43-04:00" level=info msg="Cleaning up..." component=integration_test
time="2025-09-25T12:20:43-04:00" level=info msg="Running command" arg="[cleanup]" cmd=/usr/local/bin/oc-netobserv component=integration_test
time="2025-09-25T12:21:17-04:00" level=error msg="Wait returned error: signal: hangup" component=integration_test
•time="2025-09-25T12:21:32-04:00" level=info msg="Starting command" arg="[flows --drops]" cmd=/usr/local/bin/oc-netobserv component=integration_test
time="2025-09-25T12:22:02-04:00" level=info msg="Cleaning up..." component=integration_test
time="2025-09-25T12:22:02-04:00" level=info msg="Running command" arg="[cleanup]" cmd=/usr/local/bin/oc-netobserv component=integration_test
time="2025-09-25T12:22:23-04:00" level=error msg="Wait returned error: signal: hangup" component=integration_test
•

Ran 3 of 8 Specs in 231.953 seconds
SUCCESS! -- 3 Passed | 0 Failed | 0 Pending | 5 Skipped
PASS

Ginkgo ran 1 suite in 3m58.056686761s
Test Suite Passed
```

**Comment by memodi** [comment]
thanks @oliver-smakal

**Comment by memodi** [comment]
> Yes @memodi , you are right, the `go fmt` generates a diff - a few empty liners extra -> updated, but now again need your approval to see if it goes through the tests without any other issuesYes, you are right, the `go fmt` generates a diff - a few empty liners extra -> updated,

I invited you to netobserv github org, hopefully it will not be overwritten, if it does you'd have to raise DPP request.
